lib/vanagon/project.rb

```diff
@@ -101,6 +101,22 @@ def get_replaces
       replaces.flatten.uniq
     end
 
+    # Grabs a specific service based on which name is passed in
+    # note that if the name is wrong or there was no
+    # @component.install_service call in the component, this
+    # will return nil
+    #
+    # @param [string] name of service to grab
+    # @return [@component.service obj] specific service
+    def get_service(name)
+      @components.each do |component|
+        if component.name == name
```

Comment:
I'm not sure what you're getting at, I'm fairly certain you cannot make a component without a name https://github.com/puppetlabs/vanagon/blob/master/lib/vanagon/component.rb#L44